exchanges/poloniex/poloniex_wrapper.go

```diff
@@ -792,124 +1280,336 @@ func (e *Exchange) GetFeeByType(ctx context.Context, feeBuilder *exchange.FeeBui
+			case "ISOLATED":
+				mType = margin.Isolated
+			case "CROSS":
+				mType = margin.Multi
 			}
-
 			orders = append(orders, order.Detail{
-				OrderID:  strconv.FormatInt(resp.Data[key][i].OrderNumber, 10),
-				Side:     orderSide,
-				Amount:   resp.Data[key][i].Amount,
-				Date:     orderDate,
-				Price:    resp.Data[key][i].Rate,
-				Pair:     symbol,
-				Exchange: e.Name,
+				Type:            oType,
+				OrderID:         fOrder.OrderID,
+				Side:            orderSide,
+				Amount:          fOrder.Size.Float64(),
+				Date:            fOrder.CreationTime.Time(),
+				Price:           fOrder.Price.Float64(),
+				Exchange:        e.Name,
+				Pair:            fOrder.Symbol,
+				ReduceOnly:      fOrder.ReduceOnly,
+				Leverage:        fOrder.Leverage.Float64(),
+				ExecutedAmount:  fOrder.ExecQuantity.Float64(),
+				RemainingAmount: fOrder.Size.Float64() - fOrder.ExecQuantity.Float64(),
+				ClientOrderID:   fOrder.ClientOrderID,
+				Status:          oState,
+				AssetType:       req.AssetType,
+				LastUpdated:     fOrder.UpdateTime.Time(),
+				MarginType:      mType,
+				FeeAsset:        fOrder.FeeCurrency,
+				Fee:             fOrder.FeeAmount.Float64(),
+				TimeInForce:     fOrder.TimeInForce,
 			})
 		}
+	default:
+		return nil, fmt.Errorf("%w: %q", asset.ErrNotSupported, req.AssetType)
 	}
 	return req.Filter(e.Name, orders), nil
 }
 
-// GetOrderHistory retrieves account order information
-// Can Limit response to specific order status
-func (e *Exchange) GetOrderHistory(ctx context.Context, req *order.MultiOrderRequest) (order.FilteredOrders, error) {
-	err := req.Validate()
-	if err != nil {
-		return nil, err
+func accountTypeString(assetType asset.Item) string {
+	switch assetType {
+	case asset.Spot:
+		return "SPOT"
+	case asset.Futures:
+		return "FUTURES"
+	default:
+		return ""
 	}
+}
 
-	resp, err := e.GetAuthenticatedTradeHistory(ctx,
-		req.StartTime.Unix(),
-		req.EndTime.Unix(),
-		10000)
-	if err != nil {
-		return nil, err
+func stringToAccountType(assetType string) asset.Item {
+	switch assetType {
+	case "SPOT":
+		return asset.Spot
+	case "FUTURES":
+		return asset.Futures
+	default:
+		return asset.Empty
 	}
+}
 
-	format, err := e.GetPairFormat(asset.Spot, false)
-	if err != nil {
+// GetOrderHistory retrieves account order information
+// can Limit response to specific order status
+func (e *Exchange) GetOrderHistory(ctx context.Context, req *order.MultiOrderRequest) (order.FilteredOrders, error) {
+	if err := req.Validate(); err != nil {
 		return nil, err
 	}
-
-	var orders []order.Detail
-	for key := range resp.Data {
-		var pair currency.Pair
-		pair, err = currency.NewPairDelimiter(key, format.Delimiter)
+	switch req.AssetType {
+	case asset.Spot:
+		var orderSide string
+		switch req.Side {
+		case order.Sell, order.Buy:
+			orderSide = req.Side.String()
+		case order.UnknownSide, order.AnySide:
+		default:
+			return nil, fmt.Errorf("%w: %v", order.ErrSideIsInvalid, req.Side)
+		}
```

Comment:
🚧 Change request:
I don't think we need to handhold the user like this.
If the user provides a request with an UnknownSide or AnySide then I think we can let Poloniex define what the API behaviour is - most likely error.
```suggestion
```

Reply:
It seems like this was actioned, so I'm resolving.
Better give it a reply or emoji or something so I know.